ddp-workspace/projects/ddp-pancan/src/app/components/activity/activity.component.ts

```diff
@@ -16,7 +23,37 @@ type Block = ActivitySection['blocks'][number]
     templateUrl: './activity.component.html',
     providers: [SubmitAnnouncementService, SubmissionManager],
 })
-export class ActivityComponent extends ActivityRedesignedComponent {
+export class ActivityComponent extends ActivityRedesignedComponent implements OnInit, OnDestroy {
+    dialogsClosedSubscription: Subscription;
+
+    constructor(
+        public editDialog: MatDialog,
+        logger: LoggingService,
+        windowRef: WindowRef,
+        changeRef: ChangeDetectorRef,
+        renderer: Renderer2,
+        submitService: SubmitAnnouncementService,
+        analytics: AnalyticsEventsService,
+        participantsSearch: ParticipantsSearchServiceAgent,
+        @Inject(DOCUMENT) document: any,
+        injector: Injector
+    ) {
+        super(logger, windowRef, changeRef, renderer, submitService, analytics, participantsSearch, document, injector);
+    }
+
+    public ngOnInit(): void {
+        super.ngOnInit();
+
+        this.dialogsClosedSubscription = this.editDialog.afterAllClosed.subscribe(() => {
+            this.showEmptyNestedActivityError = !this.hasNoEmptyNestedActivity;
+            setTimeout(() => this.showEmptyNestedActivityError = !this.hasNoEmptyNestedActivity, 200); // quick and dirty solution to re-check, in case activity was deleted a bit later
```

Comment:
Seems like a weird hack.
It won't work if an activity is removed with a little delay.

Reply:
Yes, it is. In some cases removing activity happens after dialogs are closed and that's why I had to add a delay there.
In order for it to be fixed the non-hacky way I think it'll need changes in SDK

Reply:
This trick is helpless.
For example in Pancan Family History:
- add a Parent's Sibling
- open and close without editing
- an error message appears ('Please enter information for the family member above or remove the empty entry')
- wait a second and remove the Parent's Sibling instance
- the error message is still there

Reply:
This setTimeout does not help.
There should be more proper way to solve it.

Reply:
Yep, that's a con of timeout, works for me but may not for others. Anyway, I'll get rid of it